### DISCUSSION TITLE
All task instance's attempts logs are eagerly loaded in the UI. Is it intentional?

### DISCUSSION THREAD

**Discussion by mjkonarski-b** [apache/airflow#29727] [General] [answered]
Hi! I noticed that when I go to task instances' log page in the web UI, all logs for all attempts are automatically loaded in the background. Is that intentional?

I think it's not necessary to load them eagerly, because users don't always read all attempts' logs. I'd be fine with clicking on an attempt link and waiting few moments for the log to show up. Current logic also puts additional load on webserver and the logging backend. 

I also noticed in the JS code, that the author's idea seemed to be to lazily load the logs https://github.com/apache/airflow/blob/78de7dd4f0317377b78d86c8ba10d3405b288dbb/airflow/www/static/js/ti_log.js#L161

If it wasn't intentional, I can submit a PR with a change. 

![task_logs](https://user-images.githubusercontent.com/95358023/220939122-40c9e883-9ea9-4bf2-a770-ff3d55b26c08.png)

**Comment by potiuk** [comment, marked as answer]
I cannot think why it would be intentional. If you can provide a PR, that would be best.
